app/Services/OpenId/UserService.php

```diff
@@ -112,6 +123,31 @@ public function __construct
         $this->log_service = $log_service;
         $this->server_ctx = $server_ctx;
         $this->identifier_service = $identifier_service;
+        $this->client_repository = $client_repository;
+    }
+
+    private function addUserCRUDAction(User $user, $payload, string $action_type = "CREATE") {
+        $payload_json = json_encode($payload);
+        $current_user = Auth::user();
```

Comment:
@romanetar please lets use this https://github.com/OpenStackweb/openstackid/commit/f494eb10e39e31d83997c18d6682a21e7a60d74e